src/session_info2/__init__.py

```diff
@@ -41,9 +44,49 @@ def _cpu_info() -> str:
         proc = platform.processor() or None
         return f"{cpu_count()} logical CPU cores{f', {proc}' if proc else ''}"
 
+    @staticmethod
+    def _gpu_info() -> str | list[str]:
+        """Get GPU info."""
+        if platform.system() == "Windows":
+            # If the platform is Windows and nvidia-smi
+            # could not be found from the environment path,
+            # try to find it from system drive with default installation path
+            nvidia_smi = shutil.which("nvidia-smi")
+            if nvidia_smi is None:
+                postfix = "\\Program Files\\NVIDIA Corporation\\NVSMI\\nvidia-smi.exe"
+                nvidia_smi = f"{os.environ['SYSTEMDRIVE']}{postfix}"
+        else:
+            nvidia_smi = "nvidia-smi"
+
+        # Get ID, processing and memory utilization for all GPUs
+        try:
+            p = Popen(  # noqa: S603
+                [
+                    nvidia_smi,
+                    "--query-gpu=index,name,memory.total,driver_version",
+                    "--format=csv,noheader,nounits",
+                ],
+                stdout=PIPE,
+            )
+            stdout, _ = p.communicate()
+        except:  # noqa: E722
+            return "No GPU found"
```

Comment:
this is what causes the bug. Don’t return `str | list[str]` when you can return a more predictable `list[str]`

```suggestion
            return ["No GPU found"]
```